n_cjson.c

```diff
@@ -115,6 +121,21 @@ N_CJSON_PUBLIC(const char*) JVersion(void)
     return NOTE_C_STRINGIZE(N_CJSON_VERSION_MAJOR) "." NOTE_C_STRINGIZE(N_CJSON_VERSION_MINOR) "." NOTE_C_STRINGIZE(N_CJSON_VERSION_PATCH);
 }
 
+NOTE_C_STATIC char Jtolower(char c)
+{
+    // Less than A -> not a valid letter.
+    // Greater than z -> not a valid letter.
+    // Between a and Z -> not a valid letter.
+    // Between a and z (inclusive) -> already lower.
+    if (c < 'A' || c > 'z' || (c < 'a' && c > 'Z') || (c >= 'a' && c <= 'z')) {

```

Comment:
This seems overly complicated.

It becomes more obvious if you swap in numbers.

Return the letter if:
x < 65 OR x > 122 OR (x < 97 AND x > 90) OR (x >= 97 AND x <= 122)

-128 >= x < 65,
x > 122,
90 < x < 97,
97 <= x <= 122

-128 <------ 65 ______ 91 --97--122--> 127

```suggestion
    if (c < 'A' || c > 'Z') {
```

Reply:
Nah I think your change works.